cpp/dolfinx/geometry/gjk.cpp

```diff
@@ -69,90 +84,122 @@ nearest_simplex(const xt::xtensor<double, 2>& s)
     if (lbb >= 0.0 and lcc >= 0.0 and (lbb + lcc) <= 1.0)
     {
       // Calculate intersection more accurately
-      auto v = math::cross(c - a, b - a);
+      // v = (c - a)  x (b - a)
+      std::array<double, 3> dx0 = {c[0] - a[0], c[1] - a[1], c[2] - a[2]};
+      std::array<double, 3> dx1 = {b[0] - a[0], b[1] - a[1], b[2] - a[2]};
+      std::array<double, 3> v = math::cross_new(dx0, dx1);
 
       // Barycentre of triangle
-      auto p = (a + b + c) / 3.0;
+      std::array<double, 3> p
+          = {(a[0] + b[0] + c[0]) / 3, (a[1] + b[1] + c[1]) / 3,
+             (a[2] + b[2] + c[2]) / 3};
+
+      double sum = v[0] * p[0] + v[1] * p[1] + v[2] * p[2];
+      double vnorm2 = v[0] * v[0] + v[1] * v[1] + v[2] * v[2];
+      for (std::size_t i = 0; i < 3; ++i)
+        v[i] *= sum / vnorm2;
 
-      // Renormalise n in plane of ABC
-      v *= xt::sum(v * p) / xt::norm_sq(v);
-      return {std::move(s), v};
+      return {std::vector<double>(s.begin(), s.end()), v};
     }
 
     // Get closest point
-    std::size_t i = xt::argmin(xt::norm_sq(s, {1}))();
-    xt::xtensor_fixed<double, xt::xshape<3>> vmin = xt::row(s, i);
-    double qmin = xt::norm_sq(vmin)();
+    std::size_t pos = 0;
+    {
+      double norm0 = std::numeric_limits<double>::max();
+      for (std::size_t i = 0; i < s.size(); i += 3)
+      {
+        double norm = 0.0;
+        for (std::size_t k = i; k < i + 3; ++k)
+          norm += s[k] * s[k];
```

Comment:
Is there a reason for not using `transform_reduce` here?

Reply:
A bit too heavyweight.